src/search-modal/SearchUI.jsx

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable react/prop-types */
+// @ts-check
+import React from 'react';
+import {
+  MenuItem,
+  ModalDialog,
+  SelectMenu,
+} from '@openedx/paragon';
+import { Check } from '@openedx/paragon/icons';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { Configure, InfiniteHits, InstantSearch } from 'react-instantsearch';
+import { instantMeiliSearch } from '@meilisearch/instant-meilisearch';
+
+import ClearFiltersButton from './ClearFiltersButton';
+import EmptyStates from './EmptyStates';
+import SearchResult from './SearchResult';
+import SearchKeywordsField from './SearchKeywordsField';
+import FilterByBlockType from './FilterByBlockType';
+import FilterByTags from './FilterByTags';
+import Stats from './Stats';
+import messages from './messages';
+
+/** @type {React.FC<{courseId: string, url: string, apiKey: string, indexName: string}>} */
+const SearchUI = (props) => {
+  const { searchClient } = React.useMemo(() => instantMeiliSearch(props.url, props.apiKey), [props.url, props.apiKey]);
```

Comment:
This fixes the key error warning in the `InfiniteHits` component. The `Hit` object needs an objectID property, which is set after passing the `primaryKey` to the `instantMeiliSearch`

```suggestion
  const { searchClient } = React.useMemo(
    () => instantMeiliSearch(props.url, props.apiKey, { primaryKey: 'id' }),
    [props.url, props.apiKey],
  );
```

Reply:
Ah, nice! Thanks for figuring that out. I've incorporated this and the other things you pointed out - good catch :)